src/daily_output.ts

```diff
@@ -0,0 +1,377 @@
+		this.latestTimestamp = Math.max(this.latestTimestamp, node.block.timestamp);
+	};
+
+	/**
+	 * Adds +1 to the file count
+	 * @param address the address of the owner wallet of the transaction
+	 */
+	private sumFile(address: string): void {
+		this.setupWallet(address);
+		this.data.wallets[address].daily.fileCount++;
+		this.data.wallets[address].weekly.fileCount++;
+		this.data.wallets[address].total.fileCount++;
+	}
+
+	/**
+	 * Sums up the size in bytes of the file
+	 * @param address the address of the owner wallet of the transaction
+	 * @param size the data size of the file data transaction
+	 */
+	private sumSize(address: string, size: number): void {
+		this.setupWallet(address);
+		this.data.wallets[address].daily.byteCount += size;
+		this.data.wallets[address].weekly.byteCount += size;
+		this.data.wallets[address].total.byteCount += size;
+	}
+
+	/**
+	 * Will fill the data of a wallet if no present
+	 * @param address the address of whom to fill the data
+	 */
+	private setupWallet(address: string): void {
+		if (!this.data.wallets[address]) {
+			this.data.wallets[address] = {
+				daily: {
+					fileCount: 0,
+					byteCount: 0,
+					changeInPercentage: 0,
+					rankPosition: 0,
+					tokensEarned: 0
+				},
+				yesterday: {
+					fileCount: 0,
+					byteCount: 0,
+					changeInPercentage: 0,
+					rankPosition: 0,
+					tokensEarned: 0
+				},
+				weekly: {
+					fileCount: 0,
+					byteCount: 0,
+					changeInPercentage: 0,
+					rankPosition: 0,
+					tokensEarned: 0
+				},
+				lastWeek: {
+					fileCount: 0,
+					byteCount: 0,
+					changeInPercentage: 0,
+					rankPosition: 0,
+					tokensEarned: 0
+				},
+				total: {
+					fileCount: 0,
+					byteCount: 0,
+					changeInPercentage: 0,
+					rankPosition: 0,
+					tokensEarned: 0
+				}
+			};
+		}
+	}
+
+	/**
+	 * Stores the JSON output of the class instance
+	 * @throws if the JSON data is invalid
+	 */
+	public write(): void {
+		if (!this.validateDataStructure(this.data)) {
+			throw new Error(`Cannot save invalid output: ${JSON.stringify(this.data)}`);
+		}
+		writeFileSync(OUTPUT_NAME, JSON.stringify(this.data, null, '\t'));
+		console.log(`Output saved at ${OUTPUT_NAME}`);
+	}
+
+	/**
+	 * @returns the content of the JSON template file
+	 */
+	private readTemplate(): unknown {
+		const data = readFileSync(OUTPUT_TEMPLATE_NAME);
+		const dataAsString = data.toString();
+		return JSON.parse(dataAsString);
+	}
+
+	/**
+	 * @returns the content of the JSON output file
+	 */
+	public readOutputFile(): unknown {
+		const data = readFileSync(OUTPUT_NAME);
+		const dataAsString = data.toString();
+		return JSON.parse(dataAsString);
+	}
+
+	/**
+	 * Validates certain object to match the schema of the OutputData
+	 * @param data a JSON object
+	 * @returns true if the data is OutputData
+	 */
+	private validateDataStructure(data: unknown): data is OutputData {
+		const dataAsOutputData = data as OutputData;
```

Comment:
We could check the schema of each `wallets` entry

Reply:
[2b14e34](https://github.com/ardriveapp/inferno-backend/pull/8/commits/2b14e34642f979833e89d8b7e4500921af8bf048)